### ISSUE TITLE
AEL replacing the quotations in launch string

### ISSUE THREAD

**Issue by tjallvar** [Wintermute0110/plugin.program.advanced.emulator.launcher#43]
Hi.
Not sure if this is the place to go for this, but since the Kodi forums are down I'm coming here.
I finally found a way to launch amiga games through kodi using gamebase. But the issue I'm having is that AEL is moving the quotes I put into the launch string and thereby messing up the launch command.

Have a look, this is what I enter into application arguments: db=GameBase_Amiga fn="$rombase$"
what comes out is this: db=GameBase_Amiga "fn=(ROMNAME)"
(ROMNAME) is the correct filename, I just replaced it here. 

As you can see he moves the " from after = to infront of fn.
And this messes up the command line obviously.

I really hope you can have a quick look at this and fix it.

**Comment by Wintermute0110** [comment]
This is the absolutely right place for AEL issues :)

OK... it could be a problem of the shlex module and the way it splits arguments. I will need you to provide a debug log. In AEL settings, activate the DEBUG log level. Ensure Kodi DEBUG is OFF in Kodi settings. Only AEL must be on debug mode. Then, start Kodi, launch a couple of ROMs, and send the log (you can use Ronnie addon to upload the log in a very convenient way). If possible, include a ROM with spaces in its filename and another with no spaces. In the debug log I will see the OS your are using and all relevant information.

**Comment by tjallvar** [comment]
Hi.
The problem only arises when romnames have spaces. Else he isn't using the quotes.
db=GameBase_Amiga fn=4-Get-It.zip <- example output of no spaces.
I uploaded the log of a few game starts to pastebin here: https://pastebin.com/TtKjUi7u
Last one works fine because of no spaces. the others fail because of the before mentioned output error.

**Comment by tjallvar** [comment]
BTW: While I have your attention, is there no way to scrape all gams in a collection? Do you have to do each one manually?

**Comment by Wintermute0110** [comment]
> BTW: While I have your attention, is there no way to scrape all gams in a collection? Do you have to do each one manually?

Currently there is no option for that. The rationale is that you scrape your launchers, then you add them to collections with metadata and artwork already scraped and finally you use game-by-game scraping in the collection for fine tuning. I may add an option to scrape all games and Favourites and ROM Collections for 0.9.8.

**Comment by tjallvar** [comment]
How do I "scrape my launchers"? any tool that can do that out of kodi? that is compatible with AEL.
Anyway, this part wasn't really the issue at hand, but its nice to have it confirmed that there is no "scrape all", so I don't have to keep searching. And yes, please please add a "scrape all" option! ;)

**Comment by Wintermute0110** [comment]
> How do I "scrape my launchers"?

Currently, batch scraping of all ROMS in a launcher is only supported by the ROM Scanner. Also, you have to make sure you have the correct settings. In AEL settings, check the "Metadata scan policy" and "Asset scan policy". By default AEL uses the offline scraper (an internal database with metadata of thousands of ROMs, very fast) and searches for local artwork.

For big collections I recommend you download artwork collections. Just place the artwork in the appropriate directories and then use the ROM Scanner or go to the launcher, open context menu, "Edit Launcher", "Manage ROMs", "Rescan local artwork" (I'm writing from memory, actual menu names may be a little bit different.

> Any tool that can do that out of kodi? that is compatible with AEL.

AEL can use any assets (images, PDFs, videos) and the only condition is that the filename must be the same as the ROM file name. Note that most artwork collections you can download follow the No-Intro or Redump ROM naming schemes.

I'm not sure... to be honest in my setup I almost don't use the scrapers: I always download artwork collections and just use local artwork. My ROM Collections (Castlevania games, Mario games, Sonic games, Pokemon games, etc.) have a maximum of 15/20 ROMs and I scrape them ROM by ROM. Having said that, I think any other front end that scrapes images may be used to download the artwork (Launchbox, EmulationStation, etc.). Then, you copy the artwork to the AEL directories and execute "Rescan local artwork". You may even use several front ends and configure same artwork directories to share artwork among several front ends.

> And yes, please please add a "scrape all" option! ;)

Will think about it. However, it is discouraged because AEL supports vasts collections and if you scrape assets online it can take a really long time. For example, in my setup I have about 30.000 in total. Scraping all assets for that online may take weeks. It is much faster to download assets collections readily available. In AEL thread first post you have links to some places where you can download artwork for your launchers. Wait until forums are online again and have a look.

**Comment by tjallvar** [comment]
I get this error when rescaning for assets: 
no-intro/redump dat not configured and audit_pclone_assets is true. Cancelling looking for assets in the Parent/Clone group.

I get that with the no-intro setting on or off.  
I can see that I'm a noob in the emul scene, just started setting up an emulator box a few days ago..

Anyway.. this is a secondary problem, if I can't launch the games from kodi, then no point scraping them there either ;)
Any luck with the actual problem at hand?

**Comment by Wintermute0110** [comment]
The message "No-intro/redump dat not configured ..." is not an error, just an information message. There is an option in AEL settings to search artwork in the Parent/Clone group. Many artwork collections (notably Hyperspin ones) only include artwork for the USA version of the ROM. If you use a No-Intro DAT file, AEL creates the Parent/Clone groups. For example: "Sonic The Hedgehog (USA)", "Sonic The Hedgehog (Europe)", "Sonic The Hedgehog (Japan)" belong to the same PClone group because are regional version of the same game. If you have artwork for the USA region, AEL is able to use that artwork for the Europe and Japan ROMs if Europe and Japan artwork is missing.

By default the option to search for artwork in the PClone group is enable by default, and because you don't have a No-Intro DAT configured for that launcher hence the message. Currently the message is a little bit cryptic... I will improve it a bit.

Also, note that AEL is in heavy development. Lots of things are changing from version to version to make a better front-end. I didn't have time yet to write proper documentation because: a) writing good documentation takes a lot of time, comparable to development time. b) Features change a lot from version to version. I don't want to spent time writing docs and tutorials and then have to change them when new features arrive.

In any case, eventually AEL feature set will stabilise and then I will improve the wiki with tutorial and setp-by-step guides. Also, AEL has a lot of features for advanced users but the defaults should work fine and be easy to use for any user. I put a lot of effort on this last point. (And yes... having a complete collection of 20/30 launchers for the most common systems, each of them with 1.000 ROMs with all artwork, bells and whistles is considered for advanced users).

**Comment by tjallvar** [comment]
Been reading up on the different types of collections. But they are mostly for consoles, not the Amiga as far as I can find. Only Tosec, and that thing has 10x different versions of the same game/program. Kinda redundant (I still keep it though hehe). 

I'm trying to "integrate" the gamebase amiga collection (games, gfx and extras) into AEL, and failing miserably. :) But again, this is something I will figure out eventually. I just need to know that I can launch the games from kodi, using AEL. But you haven't mentioned if you have any idea how to fix the argument problem?

But I thank you for taking time time to explain.

**Comment by Wintermute0110** [comment]
> Been reading up on the different types of collections. But they are mostly for consoles, not the Amiga as far as I can find. Only Tosec, and that thing has 10x different versions of the same game/program. Kinda redundant (I still keep it though hehe).

Yes, unfortunately No-Intro or Redump (equivalent of No-Intro for CD-ROM-based platforms) is not available fro Amiga games. I praise TOSEC for its huge effort in cataloging every ROM on the Internet, but I don't want to have more than 100 version of SNES Super Mario World, 99% of which are pure garbage.

> I'm trying to "integrate" the gamebase amiga collection (games, gfx and extras) into AEL, and failing miserably. :) But again, this is something I will figure out eventually. I just need to know that I can launch the games from kodi, using AEL. But you haven't mentioned if you have any idea how to fix the argument problem?

Fixing a bug takes some time and currently I'm very busy with real life. I guess you want to use WinUAE to launch your Amiga games, right? Instead of launching GameBase, why you don't try to launch WinUAE with AEL and bypass GameBase? You can use [this instructions](https://fs-uae.net/command-line) to configure FS-UAE emulator with AEL.

**Comment by tjallvar** [comment]
Haha, yeah, I know. It seems a bit overkill to keep every single version of a game, the only difference being the cracktro or simply release group.. 

Already been through launching with winuae directly. But it can't handle multiple disk games automatically.
GameBase is able to load all the disks automatically. So no disk swapping (unless more than 4 disks), but he also puts the rest of the disks in the disk swapper automatically, so u just have to click "end+#disk".

**Comment by Wintermute0110** [comment]
@ghost What happened with this? Is this bug still not fixed?
